Sources/Foundation/Thread.swift

```diff
@@ -391,26 +391,31 @@ open class Thread : NSObject {
         return backtraceAddresses { (addresses, count) in
           var symbols: [String] = []
 
-          var buffer: UnsafeMutablePointer<Int8> =
-              UnsafeMutablePointer<Int8>
-                  .allocate(capacity: MemoryLayout<SYMBOL_INFO>.size + 128)
-          defer { buffer.deallocate() }
+          let addresses: UnsafeMutableBufferPointer<PVOID?> =
+              UnsafeMutableBufferPointer<PVOID?>(start: addresses, count: count)
+          withUnsafeTemporaryAllocation(byteCount: MemoryLayout<SYMBOL_INFO>.size + 127,
+                                        alignment: 8) { buffer in
+            let pSymbolInfo: UnsafeMutablePointer<SYMBOL_INFO> =
+                buffer.baseAddress!.assumingMemoryBound(to: SYMBOL_INFO.self)
+
+            for address in addresses {
+              pSymbolInfo.pointe.SizeOfStruct =
+                    ULONG(MemoryLayout<SYMBOL_INFO>.size)
+              pSymbolInfo.pointee.MaxNameLen = 128
 
-          buffer.withMemoryRebound(to: SYMBOL_INFO.self, capacity: 1) {
-            $0.pointee.SizeOfStruct = ULONG(MemoryLayout<SYMBOL_INFO>.size)
-            $0.pointee.MaxNameLen = 128
-
-            var address = addresses
-            for _ in 1...count {
               var dwDisplacement: DWORD64 = 0
-              if !SymFromAddr(hProcess, unsafeBitCast(address.pointee, to: DWORD64.self), &dwDisplacement, $0) {
-                symbols.append("\($0.pointee)")
+              if SymFromAddr(hProcess, DWORD64(UInt(bitPattern: address)),
```

Comment:
This may need to be revisited for 32-bit?

Reply:
Oh, to use `UInt64` always ... yes, that is something that we will need to adjust.  I think that for 5.7 that is less critical as I doubt that we will get the last fixes for the 32-bit cleanliness in (it will require reworking SILGen).  Let's do that on main though.